src/Microsoft.Framework.ConfigurationModel.Xml/XmlConfigurationExtension.cs

```diff
@@ -7,7 +10,23 @@ public static class XmlConfigurationExtension
     {
         public static IConfigurationSourceRoot AddXmlFile(this IConfigurationSourceRoot configuration, string path)
         {
-            configuration.Add(new XmlConfigurationSource(path));
+            return AddXmlFile(configuration, path, false);
+        }
+
+        public static IConfigurationSourceRoot AddXmlFile(this IConfigurationSourceRoot configuration, string path, bool ignoreMissingFile)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentException(Resources.Error_InvalidFilePath, "path");
+            }
+
+            var fullPath = PathResolver.ResolveAppRelativePath(path);
+
+            if (!ignoreMissingFile || File.Exists(fullPath))
```

Comment:
This adds an extra File.Exists check that we didn't have before on all code paths. Should we check existence here and throw? Right now we let it flow all the way to the load and then you get the IO exception. It should make our stack trace better as well, since at the moment I think it  shows the `new Configuration()` as the line of code with an error instead of the AddJsonFile. If not then perhaps this should be:

```
if (ignoreMissingFile && !File.Exists(fullPath))
{
    return configuration;
}
configuration.Add(...);
```
Don't know if we really care too much, or if @davidfowl hates multiple returns or not :)